enhancements/olm-toggle-copied-csvs.md

```diff
@@ -0,0 +1,199 @@
+
+### Goals
+
+- Allieviate the resource utilization incurred by Copied CSVs
+- Don't block upgrades for customers of OpenShift
+
+### Non-Goals
+
+- Provide a replacement for Copied CSVs
+
+## Proposal
+
+Provide a CRD for OLM that allows users to enable and disable Copied CSVs for CSVs installed in `AllNamespace` mode, then for each non-Copied CSV, a single [Operator](https://github.com/operator-framework/enhancements/blob/master/enhancements/simplify-olm-apis.md#proposal) resource will be generated. This cluster-scoped resource will take the place of any related Copied CSVs.
+
+### User Stories
+
+#### Toggle Copied CSVs Off
+
+As a Cluster Admin, I want a way to disable Copied CSVs so that I can reduce OLM's resource consumption on my cluster.
+
+#### Toggle Copied CSVs On
+
+As a Cluster Admin, I want a way to enable Copied CSVs if I've reconfigured the cluster in ways that mitigate the problem; e.g. uninstalled operators from `AllNamespace` `OperatorGroups`.
+
+### Implementation Details/Notes/Constraints
+
+#### Operator Cardinality
+
+Today, Operator resources are generated on a per-Subscription basis. Since CSVs -- and thus Copied CSVs -- can exist without matching Subscriptions, OLM's controllers will be extended to generate additional Operator resources for these "free CSVs" such that:
+
+- there exists at least one`Operator` resource for every CSV
+- there exists at least one `Operator` resource for every `Subscription`
+
+That is, all CSVs installed by a given `Subscription` share an `Operator` resource __and__ any CSVs that weren't installed by a `Subscription` are related to their own `Operator` resource.
+
+#### Orphaned CSVs
+
+Orphaned CSVs are CSVs that were installed via a `Subscription` that no longer exists. Since the lifetimes -- i.e. the period of existence on a cluster --  of `Operator` resources aren't directly tied to the lifetimes of the resources that triggered their generation, the `Operator` resource(s) associated with Orphaned CSVs will continue to exist _after_ a `Subscription` is deleted. If that `Operator` resource is deleted explicitly, OLM will generate another as per the rules [outlined above](#operator-cardinality).
```

Comment:
and presumably the Operator resource will be deleted when all the CSVs associated with it go away?

Reply:
No -- it was an explicit design decision for "componentless" Operator resources to stick around to:
- support situations where Operator resources are created manually
- prevent eventual consistency from causing controllers to flap Operators; i.e. if components exist but have yet to be noticed up by the Operator controller, the respective Operator shouldn't be deleted only to be created again once components are noticed

"component" = resource associated with a given Operator resource